quote-server/server.js

```diff
@@ -1,32 +1,59 @@
-// server.js
-// This is where your node app starts
-
-//load the 'express' module which makes writing webservers easy
 import express from "express";
-//load the quotes JSON
-import quotes from "./quotes.json" assert { type: "json" };
+import crypto from "crypto";
+import cors from "cors";
 
 const app = express();
-// Now register handlers for some routes:
-//   /                  - Return some helpful welcome info (text)
-//   /quotes            - Should return all quotes (json)
-//   /quotes/random     - Should return ONE quote (json)
-app.get("/", (request, response) => {
-  response.send("Neill's Quote Server!  Ask me for /quotes/random, or /quotes");
+
+let quotes = [];
+
+app.use(cors());
+
+app.use(async (req, res, next) => {
+  const response = await fetch("https://api.quotable.io/quotes?limit=150");
+  const data = await response.json();
+  quotes = data.results;
+  next();
+});
+
+app.get("/", (req, res) => {
+  res.send(
+    "Fikret's Quote Server! Ask me for /quotes, /quotes/random or /quotes/search?term=(something)"
+  );
 });
 
 //START OF YOUR CODE...
+app.get("/quotes", (req, res) => {
+  res.json({ quotes });
```

Comment:
Does this return an array or an object? What are the advantages/disadvantages of each option? Why did you choose the one you did? (Same question about all of your endpoints :))

Reply:
It returns an object with quote property which has the value of quotes object. When someone fetches this path, they will get the data with name property that explains what the data is. Also any other information can be given with other properties. By this way our server will be easier to read and use. I chose this way because it is more instructive and user-friendly. In the other hand if I dont use {} to wrap my array, data will be unknown, user needs to know what it inside the response in advance. The advantage of this way could be privacy reasons or straight usage of the data provided. This way is more efficient and straight to achieve goal.